sycl/include/sycl/ext/oneapi/reduction.hpp

```diff
@@ -94,13 +94,11 @@ using IsReduOptForFastReduce =
                    sycl::detail::IsMaximum<T, BinaryOperation>::value)>;
 #endif
 
-// std::tuple seems to be a) too heavy and b) not copyable to device now
-// Thus sycl::detail::tuple is used instead.
-// Switching from sycl::device::tuple to std::tuple can be done by re-defining
-// the ReduTupleT type and makeReduTupleT() function below.
-template <typename... Ts> using ReduTupleT = sycl::detail::tuple<Ts...>;
+// Using std::tuple allows usage of functions like std::tuple_element_t
```

Comment:
This comment might be outdated.  I was using another approach to filter before, and I was seeing problems with `sycl::detail::tuple`.  I didn't try converting back. I just tried now, and see some problems -- I'd appreciate your thoughts.

If I swap things back to `sycl::detail::tuple` my tests no longer compile.  I think this is because the code ends up using a mix of `std::tuple` and `sycl::detail::tuple` -- I didn't realize that there are parts of the original implementation that always use `std::tuple` directly (e.g. see [`createReducers`](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/reduction.hpp#L1513-L1520)).

I could try changing my code to use a `std::tuple` for reducers instead of a `sycl::detail::tuple`, or I could try changing the original code to use a `sycl::detail::tuple` for reducers.  Assuming that both work, do you have a preference?

Reply:
detail::tuple had some pros: it did not depend on OS/host implementation, was same/predictable for windows/linux.
But still, I like the idea of using std::tuple more than supporting a custom tuple.

The only/last concern is that comment about code-efficiency. It should be verifiable by compiling a kernel with N reductions (N >=2) using a) existing compiler and b) compiler that has only 1 change - switch to std::tuple.

Reply:
Turns out it was actually very easy to switch back to `sycl::detail::tuple`, so I've done that in cb030f279bbbc962f2d1231c7701fdc448898929.  I agree we should explore replacing this with `std::tuple`, but if it's okay with you I think we should leave that to a separate PR so we can limit the scope of these changes to strictly what's necessary to support spans.